ft/txn/txn_manager.cc

```diff
@@ -538,7 +538,7 @@ void toku_txn_manager_handle_snapshot_create_for_child_txn(
         inherit_snapshot_from_parent(txn);
     }
    
-    toku_debug_txn_sync(pthread_self());
+    toku_debug_txn_sync(reinterpret_cast<uint64_t>(pthread_self()));
```

Comment:
IMHO this is not a correct fix: the issue is that FT code uses uint64_t type instead of portable pthread_t. According to the standard, pthread_t might be a struct and should be treated opaquely. If possible, the right fix would be to replace uin64_t thread-id holder types with pthread_t, to replace equality comparisons with pthread_equal calls etc.